lib/phoenix_component.ex

```diff
@@ -573,12 +550,105 @@ defmodule Phoenix.Component do
     prefixes
   end
 
-  @doc """
-  Declares a slot. See `slot/3` for more information.
-  """
-  defmacro slot(name, opts \\ []) when is_atom(name) and is_list(opts) do
-    {block, opts} = Keyword.pop(opts, :do, nil)
+  @doc ~S'''
+  Declares a function component slot.
+
+  ## Arguments
+
+  * `name` - an atom defining the name of the slot. Note that slots cannot define the same name 
+  as any other slots or attributes declared for the same component.
+  * `opts` - a keyword list of options. Defaults to `[]`.
+  * `block` - a code block containing calls to `Phoenix.Component.attr/3`. Defaults to `nil`.
+
+  ### Options
+
+  * `:required` - marks a slot as required. If a caller does not pass a value for a required slot, 
+  a compilation warning is emitted. Otherwise, an omitted slot will default to `[]`.
+  * `:doc` - documentation for the slot. Any slot attributes declared
+  will have their documentation listed alongside the slot.
+
+  ### Slot Attributes
 
+  A named slot may declare attributes by passing a block with calls to `Phoenix.Component.attr/3`.
+
+  Unlike attributes, slot attributes cannot accept the `:default` option. Passing one
+  will result in a compile warning being issued.
+
+  ### The Default Slot
+
+  The default slot can be declared by passing `:inner_block` as the `name` of the slot.
+
+  Note that the default slot cannot accept a block. Passing one will result in a compilation
+  warning being emitted.
+
+  ## Compile-Time Validations
+
+  LiveView performs some validation of slots via the `:phoenix_live_view` compiler. 
+  When slots are defined, LiveView will warn at compilation time on the caller if:
+
+  * A required slot of a component is missing.
+
+  * An unknown slot is given.
+
+  * An unknown slot attribute is given.
+
+  On the side of the function component itself, defining attributes provides the following 
+  quality of life improvements:
+
+  * Slot documentation is generated for the component.
+
+  * Calls made to the component are tracked for reflection and validation purposes.
+
+  ## Documentation Generation
+
+  Public function components that define slots will have their docs 
+  injected into the function's documentation, depending on the value 
+  of the `@doc` module attribute:
+
+  * if `@doc` is a string, the slot docs are injected into that string.
+    The optional placeholder `[[INJECT LVDOCS]]` can be used to specify where
+    in the string the docs are injected. Otherwise, the docs are appended
+    to the end of the `@doc` string.
+
+  * if `@doc` is unspecified, the slot docs are used as the
+    default `@doc` string.
+
+  * if `@doc` is `false`, the slot docs are omitted entirely.
+
+  The injected slot docs are formatted as a markdown list:
+
+    * `name` (required) - slot docs. Accepts attributes:
+      * `name` (`:type`) (required) - attr docs. Defaults to `:default`.
```

Comment:
I think this is supposed to be indented but I'm not 100% sure.

Reply:
yes indent list items to spaces please

Reply:
This was originally in a code block like this:
```markdown
  * `name` (required) - slot docs. Accepts attributes:
    * `name` (`:type`) (required) - attr docs. Defaults to `:default`.
```

Note that the second item is intended to be rendered as a child of the first, since it is a slot attribute. Does that make sense?

Reply:
Thats what I thought, it just looked strange and I wanted to gut check. Thanks!